File: WEEK1/DAY3/Prog2/P2.c
#include<stdio.h>

int reverse(int N){
    int rev=0;
    int dig=0;
    while(N>0){
        dig=N%10;
        rev=(rev*10)+dig;
        N/=10;
    }
    return rev;
}

int main(){
    int N;
    printf("Check if N is a palindrome\n");
    printf("Enter N: ");
    scanf("%d",&N);
    if(N<0){
        printf("Not palindrome\n");
    }
    else{
        if(reverse(N)==N){
            printf("Palindrome\n");
        }
        else{
            printf("Not palindrome\n");
        }
    }
    return 0;
}

File: WEEK1/DAY3/Prog2/P2.java
import java.util.*;
public class P2 {
    public static int reverse(int N){
        int rev=0;
        int dig;
        while (N>0) {
            dig=N%10;
            rev=(rev*10)+dig;
            N/=10;
        }
        return rev;
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Check if N is a palindrome");
        System.out.print("Enter N: ");
        int N=sc.nextInt();
        if(N<0){
            System.out.println("Not palindrome");
        }
        else{
            if(reverse(N)==N){
                System.out.println("Palindrome");
            }
            else{
                System.out.println("Not palindrome");
            }
        }
        sc.close();
    }
}
